src/pages/RealmExplorer.js

```diff
@@ -3,201 +3,214 @@
-      // eslint-disable-next-line no-restricted-syntax
-      for (const [key] of objects) {
-        let itemString =
-          item[key] &&
-          (typeof item[key] === 'string' ||
-            typeof item[key] === 'number' ||
-            typeof item[key].getMonth === 'function') &&
-          String(item[key]);
-        if (!itemString && item[key] && item[key].length) {
-          itemString = item[key].length;
-        }
-        if (typeof item[key] === 'boolean') {
-          itemString = item[key] ? 'True' : 'False';
-        }
-        if (!itemString && item[key] && item[key].id) itemString = item[key].id;
-        cells.push(
-          <Cell key={key} style={globalStyles.cell} textStyle={globalStyles.text} width={1}>
-            {itemString}
-          </Cell>
-        );
-      }
-    }
-    return <Row style={globalStyles.row}>{cells}</Row>;
-  }
+  const onFilterChange = filterString => {
+    setState({ ...state, filterString });
+  };
+
+  useEffect(() => setState(getUpdatedState(database, state)), [
+    state.searchString,
+    state.filterString,
+  ]);
 
-  render() {
-    const { dataSource } = this.state;
-
-    return (
-      <View style={[globalStyles.container]}>
-        <SearchBar onChange={this.onSearchChange} placeholder="Table Name" />
-        <SearchBar onChange={this.onFilterChange} placeholder="Filter" />
-        <DataTable
-          style={globalStyles.container}
-          listViewStyle={globalStyles.container}
-          dataSource={dataSource}
-          renderRow={this.renderRow}
-          renderHeader={this.renderHeader}
-        />
+  const objectFields = OBJECT_FIELDS[state.objectString];
+
+  const headerRowKeyExtractor = (_, index) => index.toString();
+  const rowKeyExtractor = ({ id }) => id;
+
+  const renderHeaderRow = headerRow => {
+    const { item } = headerRow;
+    const cells = Object.keys(item).map(columnKey => (
+      <View style={styles.cell}>
+        <Text style={styles.cellText}>{columnKey}</Text>
       </View>
-    );
-  }
-}
+    ));
+    return <View style={styles.row}>{cells}</View>;
+  };
+
+  const renderRow = row => {
+    const { item } = row;
+    const cells = Object.entries(objectFields).map(([columnKey, columnType]) => {
+      const cell = item[columnKey];
+      const cellValue = parseCell(cell, columnType);
+      return (
+        <View style={styles.cell}>
+          <Text style={styles.cellText}>{cellValue}</Text>
+        </View>
+      );
+    });
+    return <View style={styles.row}>{cells}</View>;
+  };
+
+  const headerData = [objectFields];
+  const data = state.filteredData;
+
+  return (
+    <View style={[globalStyles.container]}>
+      <SearchBar onChange={onSearchChange} placeholder="Table name" />
+      <SearchBar onChange={onFilterChange} placeholder="Filter string" />
+      <FlatList
+        data={headerData}
+        keyExtractor={headerRowKeyExtractor}
+        renderItem={renderHeaderRow}
+      />
+      <VirtualizedList
```

Comment:
flat list is a wrapper around virtualized list - they do the same thing, except `FlatList` has a simpler API and is simpler to use in general. Just use one or the other https://facebook.github.io/react-native/docs/flatlist.html#listheadercomponent

Reply:
Went for `VirtualizedList`, just for consistency with `DataTable`. `FlatList` header replaced with `ListHeaderComponent` prop. b56a04041e336171188728ccea36b0be8db36eaf.